day2/index.js

```diff
@@ -0,0 +1,101 @@
+$(() => {
+  var skillPoint = 50;
+  var str = 0;
+  var dex = 0;
+  var int = 0;
+  var lux = 0;
+
+  // STR
+  $('#minus-str').on('click', () => {
+    if (skillPoint < 50 && str > 0) {
+      skillPoint++;
+      str--;
+      updateComponent(skillPoint, '#str-progress', str);
+    }
+  });
+
+  $('#plus-str').on('click', () => {
+    if (skillPoint > 0 && str < 50) {
+      skillPoint--;
+      str++;
+      updateComponent(skillPoint, '#str-progress', str);
+    }
+  });
+
+  // DEX
+  $('#minus-dex').on('click', () => {
+    if (skillPoint < 50 && dex > 0) {
+      skillPoint++;
+      dex--;
+      updateComponent(skillPoint, '#dex-progress', dex);
+    }
+  });
+
+  $('#plus-dex').on('click', () => {
+    if (skillPoint > 0 && dex < 50) {
+      skillPoint--;
+      dex++;
+      updateComponent(skillPoint, '#dex-progress', dex);
+    }
+  });
+
+  // INT
+  $('#minus-int').on('click', () => {
+    if (skillPoint < 50 && int > 0) {
+      skillPoint++;
+      int--;
+      updateComponent(skillPoint, '#int-progress', int);
+    }
+  });
+
+  $('#plus-int').on('click', () => {
+    if (skillPoint > 0 && int < 50) {
+      skillPoint--;
+      int++;
+      updateComponent(skillPoint, '#int-progress', int);
+    }
+  });
+
+  // LUX
+  $('#minus-lux').on('click', () => {
+    if (skillPoint < 50 && lux > 0) {
+      skillPoint++;
+      lux--;
+      updateComponent(skillPoint, '#lux-progress', lux);
+    }
+  });
+
+  $('#plus-lux').on('click', () => {
+    if (skillPoint > 0 && lux < 50) {
+      skillPoint--;
+      lux++;
+      updateComponent(skillPoint, '#lux-progress', lux);
+    }
+  });
+
+  $('#random').on('click', () => {
+    skillPoint = 50;
+    while (skillPoint != 0) {
+      skillPoint = 50;
```

Comment:
In web development, we should **never** use loops inside our main thread. Because loops will block the thread and user's interaction, until the loop is finished, and we don't want users to wait.

Result of `console.log(str, dex, int, lux)`
<img width="545" alt="screen shot 2560-12-18 at 02 33 52" src="https://user-images.githubusercontent.com/9304909/34083213-29601752-e39f-11e7-8a73-1e3283d07889.png">

Try other approaches without loops. Here is my code, you can use it if you understand it already.
```
    str = Math.floor(Math.random() * skillPoint)
    skillPoint -= str
    dex = Math.floor(Math.random() * skillPoint)
    skillPoint -= dex
    int = Math.floor(Math.random() * skillPoint)
    skillPoint -= int
    lux = skillPoint
    skillPoint -= lux
```

Reply:
Your solution is very good and thank for suggestion.